compiler/luci/pass/src/FuseInstanceNormPassV2.cpp

```diff
@@ -0,0 +1,314 @@
+  CHECK_OR_FALSE(fill(&reshape_of_ifm, &mul_gamma).with_commutative_args_of(mul_as_scaled_reshape));
+
+  ifm = reshape_of_ifm->tensor();
+
+  CHECK_OR_FALSE(loco::shape_known(ifm));
+  auto ifm_shape = loco::shape_get(ifm);
+  CHECK_OR_FALSE(ifm_shape.domain() == loco::Domain::Tensor);
+  auto ifm_tensor_shape = ifm_shape.as<loco::TensorShape>();
+  CHECK_OR_FALSE(ifm_tensor_shape.rank() == 4);
+  uint32_t ifm_channel_depth = ifm_tensor_shape.dim(3).value();
+
+  CHECK_OR_FALSE(fill(&rsqrt, &const_as_gamma).with_commutative_args_of(mul_gamma));
+  CHECK_OR_FALSE(is_quasi_1D_with_dummy_dim(const_as_gamma, ifm_channel_depth));
+
+  add_as_variance = dynamic_cast<luci::CircleAdd *>(rsqrt->x());
+  CHECK_OR_FALSE(add_as_variance);
+
+  CHECK_OR_FALSE(
+      fill(&mean_as_variance, &const_as_epsilon).with_commutative_args_of(add_as_variance));
+
+  CHECK_OR_FALSE(const_as_epsilon->dtype() == loco::DataType::FLOAT32);
+  // TODO Support regarding broadcast
+  CHECK_OR_FALSE(const_as_epsilon->size<loco::DataType::FLOAT32>() == 1);
+
+  CHECK_OR_FALSE(is_instance_mean(mean_as_variance));
+  sqdiff = dynamic_cast<luci::CircleSquaredDifference *>(mean_as_variance->input());
+  CHECK_OR_FALSE(sqdiff);
+
+  loco::Node *reshape_should_be = nullptr;
+  CHECK_OR_FALSE(fill(&reshape_should_be, &mean_of_reshape).with_commutative_args_of(sqdiff));
+  CHECK_OR_FALSE(reshape_of_ifm == reshape_should_be);
+  CHECK_OR_FALSE(is_instance_mean(mean_of_reshape));
+  CHECK_OR_FALSE(reshape_of_ifm == mean_of_reshape->input());
+
+  const_as_beta = dynamic_cast<luci::CircleConst *>(sub->x());
+  CHECK_OR_FALSE(const_as_beta);
+  CHECK_OR_FALSE(is_quasi_1D_with_dummy_dim(const_as_beta, ifm_channel_depth));
+
+  mul_as_scaled_mean = dynamic_cast<luci::CircleMul *>(sub->y());
+  CHECK_OR_FALSE(mul_as_scaled_mean);
+
+  luci::CircleMul *mul_gamma_should_be = nullptr;
+  luci::CircleMean *mean_of_reshape_should_be = nullptr;
+  CHECK_OR_FALSE(fill(&mul_gamma_should_be, &mean_of_reshape_should_be)
+                     .with_commutative_args_of(mul_as_scaled_mean));
+  CHECK_OR_FALSE(mul_gamma == mul_gamma_should_be);
+  CHECK_OR_FALSE(mean_of_reshape == mean_of_reshape_should_be);
+#undef CHECK_OR_FALSE
+  _matched = true;
+  return true;
+}
+
+/**
+ * Instance norm pattern would be fused like following diagram:
+ *
+ *    [In] --------------------------- CircleInstanceNorm --- [Out]
+ *                                     / /
+ *    const_as_gamma --- TFLReshape --- /
+ *                                     /
+ *    const_as_beta ---- TFLReshape ---
+ *
+ * Note
+ *  - 'const_as_gamma' and 'const_as_beta' are from original graph
+ *  - Value of 'const_as_epsilon' would be copied to CircleInstanceNorm's attribute
+ *  - TFLReshape is added as CircleInstanceNorm only accept 1D tensor
+ *  - 'CircleConst --- TFLReshape' is expected to be fused in constant folding for Reshape
+ */
+void fuse_instance_norm(const InstanceNormPattern &p)
```

Comment:
Can you provide `fuse_instance_norm` in existing `FuseInstanceNormPass` ? 
I think it would be better not to provide new `FuseInstanceNormPassV2`. This introduces more branchs and codes.